### PR TITLE
157 Wrong Status For Original Messages

### PR DESCRIPTION
**Pull request by VladislavsPerkanuks**
Status for the message is derived from the `original` and `fuzzy` flag. Explained in here: [diff](https://github.com/translate-agent/translate/pull/163/files#diff-5c855577a14bc1fb4afddc6d09d110989e0a272eb2969bb987effbb965e67f24R14-R20)

### CODE REVIEW

**Review by kaaposc — APPROVED** [review]

File: pkg/convert/convert.go [hunk 1]
```diff
@@ -32,3 +36,24 @@ func removeEnclosingBrackets(message string) string {
 	replacer := strings.NewReplacer("{", "", "}", "")
 	return replacer.Replace(message)
 }
+
+/*
+getStatus returns the status of a message based on the original and fuzzy flags and the message string.
+
+  - Scenario 1: is original -> translated
+  - Scenario 2: not original and msg is not empty -> translated
```
Comment (janishorsts) [inline pkg/convert/convert.go:44]: This is not true, we cannot rely on the message to determine its status.

File: pkg/convert/go.go [hunk 2]
```diff
@@ -70,15 +70,26 @@ func messagesFromPipeline(m pipeline.Messages, original bool) model.Messages {
 		Original: original,
 	}
 
+	getMessage := func(m pipeline.Message) string { return m.Translation.Msg }
```
Comment (VladislavsPerkanuks) [inline pkg/convert/go.go:73]: When not original, we should extract from the translation not the message.

File: pkg/convert/xliff12.go [hunk 3]
```diff
@@ -64,19 +64,24 @@ func FromXliff12(data []byte, original bool) (model.Messages, error) {
 	}
 
 	getMessage := func(t transUnit) string { return t.Source }
+	status := model.MessageStatusTranslated
 
 	// Check if a target language is set
 	if !messages.Original {
```
Comment (janishorsts) [inline pkg/convert/xliff12.go:70]: Can you please replace the negation with the following? It should improve readability.

```suggestion
	if messages.Original {
```

File: pkg/convert/xliff2.go [hunk 4]
```diff
@@ -53,11 +53,13 @@ func FromXliff2(data []byte, original bool) (model.Messages, error) {
 	}
 
 	getMessage := func(u unit) string { return u.Source }
+	status := model.MessageStatusTranslated
 
 	// Check if a target language is set
 	if !messages.Original {
```
Comment (janishorsts) [inline pkg/convert/xliff2.go:59]: The same here - replace the negation and update the flow.

**Pull request merged**